### ISSUE TITLE
Possible morphology.opening bug

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#1049]
Hi all,

In developing one of the advanced tutorials, I think I may have uncovered an error in our morphological opening operation. Here's the code:

```python
from skimage.morphology import (opening, binary_opening,
                                selem, remove_small_objects)
from scipy import ndimage as nd
strel = selem.disk(radius=2)

chrs = []
titles = []

chr0 = nd.binary_opening(chromosomes_adapt, strel)
chr1 = nd.binary_opening(chr0, strel)
chrs.append(remove_small_objects(chr1, 256))
titles.append("ndimage manual")

chr1 = nd.binary_opening(chromosomes_adapt, strel, iterations=2)
chrs.append(remove_small_objects(chr1, 256))
titles.append("ndimage iterations")

chr0 = opening(chromosomes_adapt, strel)
chr1 = opening(chr0, strel)
chrs.append(remove_small_objects(chr1, 256))
titles.append("skimage opening")

chr0 = binary_opening(chromosomes_adapt, strel)
chr1 = binary_opening(chr0, strel)
chrs.append(remove_small_objects(chr1, 256))
titles.append("skimage binary_opening")

fig, axes = plt.subplots(ncols=2, nrows=2, figsize=(15, 12))
for ax, img, title in zip(axes.ravel(), chrs, titles):
    ax.imshow(img)
    ax.set_title(title)
```

And here's the output:

![screen shot 2014-07-03 at 6 50 48 pm](https://cloud.githubusercontent.com/assets/492549/3477545/ac8e6d2e-031d-11e4-96da-778ab5b62ed4.png)

Now, never mind that the two ndimage outputs don't match even though they should, theoretically... =P The point is that the skimage ones are way off. I'm not sure who's doing it wrong here but I'm filing this bug to make sure we investigate at some point...

**Comment by stefanv** [comment]
Does your refactor of morphology handle this issue?

**Comment by jni** [comment]
Yes it does! =)

**Comment by jni** [comment]
Fixed by #1310.
